### ISSUE TITLE
When an account appears in the base and the masked layer of a Masked Merkle tree, it should only fold over the masked layered account and not the base

### ISSUE THREAD

**Issue by johnwu93** [CodaProtocol/coda#1563]
When I folded over a masked ledger, I got the following result:

```
	Address: 000000000000000000000000000000, Account: (Public Key: KImusi2ZkuWGPXN0YoOiOYRwIVLTqj91h1lYCbosEBDza7AEtmgBAAAB, Balance: 10000000) 
	Address: 000000000000000000000000000001, Account: (Public Key: KF/gCh9fJ7rsTA72qygPI+BW1hLOGmNHIj+jXsv30WJpIkrVxzsBAAAA, Balance: 100) 
	Address: 000000000000000000000000000010, Account: (Public Key: KPmGh0Hcd+reZhLFrPOhWcYMkSi4qvfkGNkO2g0i1Sz9M8IPakABAAAA, Balance: 1000) 
	Address: 000000000000000000000000000011, Account: (Public Key: KF5Lzhu3HEKR20EaK8bwKBbZKKVBRRowuYOLFIqye5MgMi4DqLADAAAA, Balance: 1000) 
	Address: 000000000000000000000000000100, Account: (Public Key: KPwcHBWoR9QzBsm/g46deD7eGFTId0n1xxBV4VoCcDBB3ik4/ekCAAAA, Balance: 1000) 
	Address: 000000000000000000000000000101, Account: (Public Key: KF4DPWVIdbXOLus9wM/HZS0rIFnX+YpGO9jnAdFt7xLR2QNB66EBAAAB, Balance: 1000) 
	Address: 000000000000000000000000000110, Account: (Public Key: KEYVC4/00VIjid9Sk7xGzZ46dQ7mWlZ5eFeSSnD7OLvw+NKq25IAAAAA, Balance: 1000) 
	Address: 000000000000000000000000000111, Account: (Public Key: KKtOMsn5QagJty8uN/9uJeA18V/4ZHWAZl2tqR391DcOBCSiPlYBAAAB, Balance: 1000) 
	Address: 000000000000000000000000001000, Account: (Public Key: KB8X1InBWdbUBnknkjz33c2MaCvzxSZqqJugHwK5cCQysb87v+AAAAAB, Balance: 1000) 
	Address: 000000000000000000000000001001, Account: (Public Key: KFhYqUSIKJHkk8ELghLu27TMwUw4zEwwdIwVn06y/z5BebexrPgBAAAA, Balance: 1000) 
	Address: 000000000000000000000000001010, Account: (Public Key: KHYImaHvbNz3wCv5w2TpJLrZmLKP5y6YcYk6grOzpdXhF4cFcxIDAAAA, Balance: 1000) 
	Address: 000000000000000000000000001011, Account: (Public Key: KGmEwVccJY/A9dYFzR16F5bmSS7r3HgArxhVLhpmyhClWeEPWoACAAAB, Balance: 1000) 
	Address: 000000000000000000000000001100, Account: (Public Key: KKjyGpCXQZzIMX71qIXrJi0plpvhTb77DBMnZBZOMospQ+7Ssh8BAAAA, Balance: 1000) 
	Address: 000000000000000000000000001101, Account: (Public Key: KKZnIR0wa7JnCOEOe+0MbPLrMNc8I9Hu22fDD/iU5tIiYS+NwnwCAAAA, Balance: 1000) 
	Address: 000000000000000000000000001110, Account: (Public Key: KK5afuhiG4JzEMGtV0Id4oe7hZKta46imNd6UHynVzhskFsDYqkAAAAA, Balance: 1000) 
	Address: 000000000000000000000000001111, Account: (Public Key: KHvST2E7CskzVqg8ZL0DuWZyRGkJrEBTN8Qif2xhrLvCifowKO8BAAAB, Balance: 1000) 
	Address: 000000000000000000000000010000, Account: (Public Key: KM7BRDKfDn7HJQZ7vtCMLvL9oBSr5rAGM0ObeHMJUQySWwY4Tw0DAAAA, Balance: 1000) 
	Address: 000000000000000000000000010001, Account: (Public Key: KNJqKTfF6H2/Z0jqKa947fdjOj0q7GkfnCW5OI6Gf6/cmdAk0b0BAAAB, Balance: 1000) 
	Address: 000000000000000000000000000110, Account: (Public Key: KEYVC4/00VIjid9Sk7xGzZ46dQ7mWlZ5eFeSSnD7OLvw+NKq25IAAAAA, Balance: 1002) 
	Address: 000000000000000000000000001011, Account: (Public Key: KGmEwVccJY/A9dYFzR16F5bmSS7r3HgArxhVLhpmyhClWeEPWoACAAAB, Balance: 1000) 
	Address: 000000000000000000000000000010, Account: (Public Key: KPmGh0Hcd+reZhLFrPOhWcYMkSi4qvfkGNkO2g0i1Sz9M8IPakABAAAA, Balance: 999) 
	Address: 000000000000000000000000001001, Account: (Public Key: KFhYqUSIKJHkk8ELghLu27TMwUw4zEwwdIwVn06y/z5BebexrPgBAAAA, Balance: 1000) 
	Address: 000000000000000000000000001010, Account: (Public Key: KHYImaHvbNz3wCv5w2TpJLrZmLKP5y6YcYk6grOzpdXhF4cFcxIDAAAA, Balance: 999) 
	Address: 000000000000000000000000000111, Account: (Public Key: KKtOMsn5QagJty8uN/9uJeA18V/4ZHWAZl2tqR391DcOBCSiPlYBAAAB, Balance: 1001) 
	Address: 000000000000000000000000001101, Account: (Public Key: KKZnIR0wa7JnCOEOe+0MbPLrMNc8I9Hu22fDD/iU5tIiYS+NwnwCAAAA, Balance: 999) 
	Address: 000000000000000000000000000101, Account: (Public Key: KF4DPWVIdbXOLus9wM/HZS0rIFnX+YpGO9jnAdFt7xLR2QNB66EBAAAB, Balance: 999) 
	Address: 000000000000000000000000001000, Account: (Public Key: KB8X1InBWdbUBnknkjz33c2MaCvzxSZqqJugHwK5cCQysb87v+AAAAAB, Balance: 1000) 
	Address: 000000000000000000000000000001, Account: (Public Key: KF/gCh9fJ7rsTA72qygPI+BW1hLOGmNHIj+jXsv30WJpIkrVxzsBAAAA, Balance: 99) 
	Address: 000000000000000000000000010000, Account: (Public Key: KM7BRDKfDn7HJQZ7vtCMLvL9oBSr5rAGM0ObeHMJUQySWwY4Tw0DAAAA, Balance: 1000) 
	Address: 000000000000000000000000000011, Account: (Public Key: KF5Lzhu3HEKR20EaK8bwKBbZKKVBRRowuYOLFIqye5MgMi4DqLADAAAA, Balance: 1003) 
	Address: 000000000000000000000000000000, Account: (Public Key: KImusi2ZkuWGPXN0YoOiOYRwIVLTqj91h1lYCbosEBDza7AEtmgBAAAB, Balance: 10000010) 
	Address: 000000000000000000000000000100, Account: (Public Key: KPwcHBWoR9QzBsm/g46deD7eGFTId0n1xxBV4VoCcDBB3ik4/ekCAAAA, Balance: 1000) 
	Address: 000000000000000000000000001100, Account: (Public Key: KKjyGpCXQZzIMX71qIXrJi0plpvhTb77DBMnZBZOMospQ+7Ssh8BAAAA, Balance: 1001) 
	Address: 000000000000000000000000001111, Account: (Public Key: KHvST2E7CskzVqg8ZL0DuWZyRGkJrEBTN8Qif2xhrLvCifowKO8BAAAB, Balance: 999) 
	Address: 000000000000000000000000010001, Account: (Public Key: KNJqKTfF6H2/Z0jqKa947fdjOj0q7GkfnCW5OI6Gf6/cmdAk0b0BAAAB, Balance: 1000) 
	Address: 000000000000000000000000001110, Account: (Public Key: KK5afuhiG4JzEMGtV0Id4oe7hZKta46imNd6UHynVzhskFsDYqkAAAAA, Balance: 999) 
	Address: 000000000000000000000000000110, Account: (Public Key: KEYVC4/00VIjid9Sk7xGzZ46dQ7mWlZ5eFeSSnD7OLvw+NKq25IAAAAA, Balance: 1001) 
	Address: 000000000000000000000000001011, Account: (Public Key: KGmEwVccJY/A9dYFzR16F5bmSS7r3HgArxhVLhpmyhClWeEPWoACAAAB, Balance: 1000) 
	Address: 000000000000000000000000010010, Account: (Public Key: KIYiwmMEzXqyNqfigusjS5qdIvaOriouDAwi8iBe+oEEpaOkDEkBAAAA, Balance: 1) 
	Address: 000000000000000000000000000010, Account: (Public Key: KPmGh0Hcd+reZhLFrPOhWcYMkSi4qvfkGNkO2g0i1Sz9M8IPakABAAAA, Balance: 999) 
	Address: 000000000000000000000000001001, Account: (Public Key: KFhYqUSIKJHkk8ELghLu27TMwUw4zEwwdIwVn06y/z5BebexrPgBAAAA, Balance: 1000) 
	Address: 000000000000000000000000001010, Account: (Public Key: KHYImaHvbNz3wCv5w2TpJLrZmLKP5y6YcYk6grOzpdXhF4cFcxIDAAAA, Balance: 999) 
	Address: 000000000000000000000000000111, Account: (Public Key: KKtOMsn5QagJty8uN/9uJeA18V/4ZHWAZl2tqR391DcOBCSiPlYBAAAB, Balance: 1002) 
	Address: 000000000000000000000000001101, Account: (Public Key: KKZnIR0wa7JnCOEOe+0MbPLrMNc8I9Hu22fDD/iU5tIiYS+NwnwCAAAA, Balance: 1000) 
	Address: 000000000000000000000000000101, Account: (Public Key: KF4DPWVIdbXOLus9wM/HZS0rIFnX+YpGO9jnAdFt7xLR2QNB66EBAAAB, Balance: 999) 
	Address: 000000000000000000000000001000, Account: (Public Key: KB8X1InBWdbUBnknkjz33c2MaCvzxSZqqJugHwK5cCQysb87v+AAAAAB, Balance: 999) 
	Address: 000000000000000000000000000001, Account: (Public Key: KF/gCh9fJ7rsTA72qygPI+BW1hLOGmNHIj+jXsv30WJpIkrVxzsBAAAA, Balance: 99) 
	Address: 000000000000000000000000010000, Account: (Public Key: KM7BRDKfDn7HJQZ7vtCMLvL9oBSr5rAGM0ObeHMJUQySWwY4Tw0DAAAA, Balance: 1000) 
	Address: 000000000000000000000000000011, Account: (Public Key: KF5Lzhu3HEKR20EaK8bwKBbZKKVBRRowuYOLFIqye5MgMi4DqLADAAAA, Balance: 1005) 
	Address: 000000000000000000000000000000, Account: (Public Key: KImusi2ZkuWGPXN0YoOiOYRwIVLTqj91h1lYCbosEBDza7AEtmgBAAAB, Balance: 10000019) 
	Address: 000000000000000000000000000100, Account: (Public Key: KPwcHBWoR9QzBsm/g46deD7eGFTId0n1xxBV4VoCcDBB3ik4/ekCAAAA, Balance: 1000) 
	Address: 000000000000000000000000001100, Account: (Public Key: KKjyGpCXQZzIMX71qIXrJi0plpvhTb77DBMnZBZOMospQ+7Ssh8BAAAA, Balance: 1000) 
	Address: 000000000000000000000000001111, Account: (Public Key: KHvST2E7CskzVqg8ZL0DuWZyRGkJrEBTN8Qif2xhrLvCifowKO8BAAAB, Balance: 1000) 
	Address: 000000000000000000000000010001, Account: (Public Key: KNJqKTfF6H2/Z0jqKa947fdjOj0q7GkfnCW5OI6Gf6/cmdAk0b0BAAAB, Balance: 999) 
	Address: 000000000000000000000000001110, Account: (Public Key: KK5afuhiG4JzEMGtV0Id4oe7hZKta46imNd6UHynVzhskFsDYqkAAAAA, Balance: 998) 
```

Here is it's base:

```
	Address: 000000000000000000000000000000, Account: (Public Key: KImusi2ZkuWGPXN0YoOiOYRwIVLTqj91h1lYCbosEBDza7AEtmgBAAAB, Balance: 10000000) 
	Address: 000000000000000000000000000001, Account: (Public Key: KF/gCh9fJ7rsTA72qygPI+BW1hLOGmNHIj+jXsv30WJpIkrVxzsBAAAA, Balance: 100) 
	Address: 000000000000000000000000000010, Account: (Public Key: KPmGh0Hcd+reZhLFrPOhWcYMkSi4qvfkGNkO2g0i1Sz9M8IPakABAAAA, Balance: 1000) 
	Address: 000000000000000000000000000011, Account: (Public Key: KF5Lzhu3HEKR20EaK8bwKBbZKKVBRRowuYOLFIqye5MgMi4DqLADAAAA, Balance: 1000) 
	Address: 000000000000000000000000000100, Account: (Public Key: KPwcHBWoR9QzBsm/g46deD7eGFTId0n1xxBV4VoCcDBB3ik4/ekCAAAA, Balance: 1000) 
	Address: 000000000000000000000000000101, Account: (Public Key: KF4DPWVIdbXOLus9wM/HZS0rIFnX+YpGO9jnAdFt7xLR2QNB66EBAAAB, Balance: 1000) 
	Address: 000000000000000000000000000110, Account: (Public Key: KEYVC4/00VIjid9Sk7xGzZ46dQ7mWlZ5eFeSSnD7OLvw+NKq25IAAAAA, Balance: 1000) 
	Address: 000000000000000000000000000111, Account: (Public Key: KKtOMsn5QagJty8uN/9uJeA18V/4ZHWAZl2tqR391DcOBCSiPlYBAAAB, Balance: 1000) 
	Address: 000000000000000000000000001000, Account: (Public Key: KB8X1InBWdbUBnknkjz33c2MaCvzxSZqqJugHwK5cCQysb87v+AAAAAB, Balance: 1000) 
	Address: 000000000000000000000000001001, Account: (Public Key: KFhYqUSIKJHkk8ELghLu27TMwUw4zEwwdIwVn06y/z5BebexrPgBAAAA, Balance: 1000) 
	Address: 000000000000000000000000001010, Account: (Public Key: KHYImaHvbNz3wCv5w2TpJLrZmLKP5y6YcYk6grOzpdXhF4cFcxIDAAAA, Balance: 1000) 
	Address: 000000000000000000000000001011, Account: (Public Key: KGmEwVccJY/A9dYFzR16F5bmSS7r3HgArxhVLhpmyhClWeEPWoACAAAB, Balance: 1000) 
	Address: 000000000000000000000000001100, Account: (Public Key: KKjyGpCXQZzIMX71qIXrJi0plpvhTb77DBMnZBZOMospQ+7Ssh8BAAAA, Balance: 1000) 
	Address: 000000000000000000000000001101, Account: (Public Key: KKZnIR0wa7JnCOEOe+0MbPLrMNc8I9Hu22fDD/iU5tIiYS+NwnwCAAAA, Balance: 1000) 
	Address: 000000000000000000000000001110, Account: (Public Key: KK5afuhiG4JzEMGtV0Id4oe7hZKta46imNd6UHynVzhskFsDYqkAAAAA, Balance: 1000) 
	Address: 000000000000000000000000001111, Account: (Public Key: KHvST2E7CskzVqg8ZL0DuWZyRGkJrEBTN8Qif2xhrLvCifowKO8BAAAB, Balance: 1000) 
	Address: 000000000000000000000000010000, Account: (Public Key: KM7BRDKfDn7HJQZ7vtCMLvL9oBSr5rAGM0ObeHMJUQySWwY4Tw0DAAAA, Balance: 1000) 
	Address: 000000000000000000000000010001, Account: (Public Key: KNJqKTfF6H2/Z0jqKa947fdjOj0q7GkfnCW5OI6Gf6/cmdAk0b0BAAAB, Balance: 1000) 
```

You can see some overlap between the two.

**Comment by psteckler** [comment]
The current code for `foldi` should ignore keys in the parent that are in the mask.

Do you have code that shows the error?

**Comment by psteckler** [comment]
Apparently, there's no bug, per @johnwu93.

**Comment by psteckler** [comment]
Re-opened, because the invalid version of `foldi` was still in the mask code. Will be fixed by #1575.
